### ISSUE TITLE
Enchancment: Add caching functionality to ExecuteQueryMultiple

### ISSUE THREAD

**Issue by BieleckiLtd** [mikependon/RepoDB#609]
Currently `ExecuteQueryMultiple `does not have overload that will take `ICache` as a parameter, same as for example `ExecuteQuery` does. It would be useful to being able to return data from cache for `ExecuteQueryMultiple` method too.

**Comment by mikependon** [comment]
The result of `ExcuteQueryMultple` is a custom object  `QueryMultpleExtractor`. Do you mean the `Extract` method will read from the cache?

**Comment by BieleckiLtd** [comment]
Is `Extract` making actual trip to the database?

**Comment by mikependon** [comment]
No, it is the ExecuteQueryMultiple, but no extraction is happening until the Extract method is called. Maybe you can play around with it and let me know the possibility on how to do the caching there?

**Comment by mikependon** [comment]
I am afraid to say it is a limitation? But let us see if this is possible.

**Comment by BieleckiLtd** [comment]
In this case I would say it must happen in `ExecuteQueryMultiple`. If database was already queried by the time `Extract` was called then it would make no sense to retrieve data from cache at this point.

**Comment by mikependon** [comment]
This operation is lazy, if we are only caching at the ExecuteQueryMultiple, there is no data to cache there, unless the Extract method is called. I think, it is a bit odd to have the caching in the Extract method and have the cache retrieval at the ExecuteQueryMultiple operation. But I suggest, you play around with it and draw a code so we can implement it.
